src/data_elements.jl

```diff
@@ -211,9 +211,77 @@ function addbattery!(elements, name, powerbalance, storagecap, lossbattery, char
     addcapacity!(elements, "DischargeCapacity_" * name, "Upper", chargecap, dischargename, FLOW_CONCEPT)
 end
 
+function elements_to_df(elements)
+    return DataFrame(
+        conceptname=[e.conceptname for e in elements], 
+        typename=[e.typename for e in elements], 
+        instancename=[e.instancename for e in elements], 
+        value = [e.value for e in elements]
+    )
+end
 
+function add_extra_col(df)
+    function add_dict_as_columns(df_input)
+        df_dict = DataFrame()
+        for d in df_input
+            df_dict = vcat(df_dict, DataFrame(d))
+        end
+        return df_dict
+    end
+    df_extra = hcat(df, add_dict_as_columns(df.value))
+    return select!(df_extra, Not(:value))
+end
 
+function check_ref(main_df, df_type, col)
+    if ~(col in ["Param", "WhichInstance", "Level", "Profile", "Commodity", "Balance", "Flow"])
```

Comment:
Does this mean that we do not validate that these references exist? Would like to do so. 

For Param, Level, Profile, Commodity you can validate the references that are String.
Whichinstance, Balance and Flow should be validated.